src/gui/torrentmodel.cpp

```diff
@@ -314,47 +298,53 @@ void TorrentModel::handleTorrentsUpdated()
 
 // Static functions
 
-QIcon getIconByState(BitTorrent::TorrentState state)
+static QString getIconId(BitTorrent::TorrentState state)
 {
     switch (state) {
     case BitTorrent::TorrentState::Downloading:
     case BitTorrent::TorrentState::ForcedDownloading:
     case BitTorrent::TorrentState::DownloadingMetadata:
-        return getDownloadingIcon();
+        return "downloading";
```

Comment:
>and I can then return a const QLatin1String & reference...

please don't over-complicate things...
returning QString is perfectly fine, either it will be optimized out by compiler RVO or it will just change QString internal pointer, either way no real copy will be performed (if not modified).